handlers/hub_test.go

```diff
@@ -0,0 +1,341 @@
+	r := router.New()
+
+	r.Default(
+		handlers.SetConfig(db, []byte(tokenSecret)),
+	)
+
+	r.GET("/api/v0/hub", handlers.Auth, handlers.DeleteHub)
+
+	return httptest.NewServer(r), nil
+}
+
+func TestAddHub(t *testing.T) {
+	// setup DB
+	db := testhelpers.SetupDB(t)
+	defer db.Close()
+
+	// setup server
+	ts, err := setupServerAddHub(db, []byte("secret"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ts.Close()
+
+	// create a user
+	u := &data.User{
+		Username: "foo",
+		Email:    "foo@example.com",
+	}
+	if err := u.EncryptPassword("password"); err != nil {
+		t.Fatal(err)
+	}
+	if err = u.Insert(db); err != nil {
+		t.Fatal(err)
+	}
+
+	// create a token for the user
+	tok := data.Token{
+		UserID:    u.ID,
+		ExpiresIn: (30 * 24 * time.Hour).Nanoseconds(), // 30 days
+	}
+	if err := tok.Insert(db); err != nil {
+		t.Fatal(err)
+	}
+
+	// get the encoded JSON Web Token
+	jwt, err := tok.EncodeJWT([]byte("secret"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	hub := data.Hub{
+		Slug:   "1234",
+		UserID: u.ID,
+	}
+	if err := hub.Insert(db); err != nil {
+		t.Fatal(err)
+	}
+
+	type testCase struct {
+		path       string
+		statusCode int
+		body       string
+	}
+
+	jsonRespRegex := regexp.MustCompile(`^{"id":2,"slug":"abcd","user_id":1,"created_at":.+,"updated_at":.+}$`)
+
+	// test when valid params are provided
+	sc := testCase{"?slug=abcd&access_token=" + jwt, http.StatusOK, `{"id":2,"slug":"abcd","user_id":1,"created_at":.+,"updated_at":.+}`}
+
+	res, err := http.Get(ts.URL + path.Join("/api/v0/hub", sc.path))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if res.StatusCode != sc.statusCode {
+		t.Errorf("%s - Expected status code %v, Got %v", sc.path, sc.statusCode, res.StatusCode)
+	}
+	b, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if body := string(b); !jsonRespRegex.MatchString(body) {
+		t.Errorf("%s - Expected response body to be %v, Got %v", sc.path, sc.body, body)
+	}
+
+	tCases := []testCase{
+		// when slug param is missing
+		{"?access_token=" + jwt, http.StatusBadRequest, `{"error":"invalid_request","error_description":"slug required"}`},
+
+		// when access_token param is missing
+		{"?slug=abcd", http.StatusUnauthorized, `{"error":"invalid_token","error_description":"no token present in request"}`},
+
+		// when trying to add existing hub
+		{"?slug=1234&access_token=" + jwt, http.StatusBadRequest, `{"error":"unique_violation","error_description":"hub exists"}`},
+	}
+	for _, tc := range tCases {
+		res, err := http.Get(ts.URL + path.Join("/api/v0/hub", tc.path))
+		if err != nil {
+			t.Fatal(err)
+		}
+		if res.StatusCode != tc.statusCode {
+			t.Errorf("%s - Expected status code %v, Got %v", tc.path, tc.statusCode, res.StatusCode)
+		}
+		b, err := ioutil.ReadAll(res.Body)
+		res.Body.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// exclude testing json body for valid params because of arbitrary timestamps from postgres.
```

Comment:
i don't think this comment is needed, since the valid params are now tested separately.